cmd/deploy.go

```diff
@@ -51,13 +51,16 @@ kn func deploy --registry quay.io/myuser
 kn func deploy --image quay.io/myuser/myfunc -n myns
 `,
 		SuggestFor: []string{"delpoy", "deplyo"},
-		PreRunE:    bindEnv("image", "namespace", "path", "registry", "confirm", "build", "push"),
+		PreRunE:    bindEnv("image", "namespace", "path", "registry", "confirm", "build", "push", "git-url", "git-branch", "git-dir"),
 	}
 
 	cmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all configuration options (Env: $FUNC_CONFIRM)")
 	cmd.Flags().StringArrayP("env", "e", []string{}, "Environment variable to set in the form NAME=VALUE. "+
 		"You may provide this flag multiple times for setting multiple environment variables. "+
 		"To unset, specify the environment variable name followed by a \"-\" (e.g., NAME-).")
+	cmd.Flags().StringP("git-url", "g", "", "Repo url to push the code to be built")
+	cmd.Flags().StringP("git-branch", "t", "", "Git branch to be used for remote builds")
+	cmd.Flags().StringP("git-dir", "d", "", "Directory in the repo where the function is located")
```

Comment:
Most of the other commands include a note about the use of environment variables such as `FUNC_GIT_URL` and friends. Perhaps consider that here.